cli/azd/pkg/infra/provisioning/bicep_provider.go

```diff
@@ -75,203 +110,350 @@ func (p *BicepProvider) Preview(ctx context.Context) *async.InteractiveTaskWithP
 
-func (p *BicepProvider) Destroy(ctx context.Context, preview *Preview) *async.InteractiveTaskWithProgress[*DestroyResult, *DestroyProgress] {
-	return async.RunInteractiveTaskWithProgress(
-		func(asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress]) {
-			destroyResult := DestroyResult{}
+// Deletes the azure resources within the deployment
+func (p *BicepProvider) destroyResourceGroups(ctx context.Context, asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress], options DestroyOptions, resourceGroups []string, allResources []azcli.AzCliResource) error {
+	if !options.Force() {
+		err := asyncContext.Interact(func() error {
+			confirmDestroy, err := p.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      fmt.Sprintf("This will delete %d resources, are you sure you want to continue?", len(allResources)),
+				DefaultValue: false,
+			})
 
-			asyncContext.SetProgress(&DestroyProgress{Message: "Fetching resource groups", Timestamp: time.Now()})
-			resourceManager := infra.NewAzureResourceManager(p.azCli)
-			resourceGroups, err := resourceManager.GetResourceGroupsForDeployment(ctx, p.env.GetSubscriptionId(), p.env.GetEnvName())
 			if err != nil {
-				asyncContext.SetError(fmt.Errorf("discovering resource groups from deployment: %w", err))
+				return fmt.Errorf("prompting for delete confirmation: %w", err)
 			}
 
-			var allResources []azcli.AzCliResource
+			if !confirmDestroy {
+				return errors.New("user denied delete confirmation")
+			}
 
-			asyncContext.SetProgress(&DestroyProgress{Message: "Fetching resources", Timestamp: time.Now()})
-			for _, resourceGroup := range resourceGroups {
-				resources, err := p.azCli.ListResourceGroupResources(ctx, p.env.GetSubscriptionId(), resourceGroup)
-				if err != nil {
-					asyncContext.SetError(fmt.Errorf("listing resource group %s: %w", resourceGroup, err))
-				}
+			return nil
+		})
 
-				allResources = append(allResources, resources...)
-			}
+		if err != nil {
+			return err
+		}
+	}
 
-			err = asyncContext.Interact(func() error {
-				confirmDestroy, err := p.console.Confirm(ctx, input.ConsoleOptions{
-					Message:      fmt.Sprintf("This will delete %d resources, are you sure you want to continue?", len(allResources)),
-					DefaultValue: false,
-				})
+	for _, resourceGroup := range resourceGroups {
+		message := fmt.Sprintf("%s resource group %s", output.WithErrorFormat("Deleting"), output.WithHighLightFormat(resourceGroup))
+		asyncContext.SetProgress(&DestroyProgress{Message: message, Timestamp: time.Now()})
 
-				if err != nil {
-					return err
-				}
+		if err := p.azCli.DeleteResourceGroup(ctx, p.env.GetSubscriptionId(), resourceGroup); err != nil {
+			return err
+		}
 
-				if !confirmDestroy {
-					return errors.New("user denied confirmation")
-				}
+		p.console.Message(ctx, fmt.Sprintf("💀 %s resource group %s", output.WithErrorFormat("Deleted"), output.WithHighLightFormat(resourceGroup)))
```

Comment:
NIT: Slight preference to just removing all the emojis but if we are to keep them, could we at least remove the skulls for now?